From 3a4368dccf60d4b6588976006172782c7ce08b44 Mon Sep 17 00:00:00 2001
From: Bjorn Andersson <bjorn.andersson@oss.qualcomm.com>
Date: Tue, 18 Mar 2025 22:17:02 -0500
Subject: [PATCH 3/6] arm64: dts: qcom: x1e80100-dell-xps13-9345: Disable
 PM8010

The Qualcomm X Elite reference design uses the PM8010 PMIC for camera
use cases, but the Dell XPS13 doesn't. Disable this PMIC to avoid the
error in the kernel log caused by an attempt to access it during boot.

Fixes: f5b788d0e8cd ("arm64: dts: qcom: Add support for X1-based Dell XPS 13 9345")
Signed-off-by: Bjorn Andersson <bjorn.andersson@oss.qualcomm.com>
---
 arch/arm64/boot/dts/qcom/x1e80100-dell-xps13-9345.dts | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/x1e80100-dell-xps13-9345.dts b/arch/arm64/boot/dts/qcom/x1e80100-dell-xps13-9345.dts
index 967f6dba0878..684441bc3eb3 100644
--- a/arch/arm64/boot/dts/qcom/x1e80100-dell-xps13-9345.dts
+++ b/arch/arm64/boot/dts/qcom/x1e80100-dell-xps13-9345.dts
@@ -867,6 +867,10 @@ &pcie6a_phy {
 	status = "okay";
 };
 
+&pm8010 {
+	status = "disabled";
+};
+
 &pm8550_gpios {
 	rtmr0_default: rtmr0-reset-n-active-state {
 		pins = "gpio10";
-- 
2.45.2

